### PR TITLE
Plaenh 2 drush migrate command

### PR DESCRIPTION
**Pull request by TudorCCalin**
# [UHF-0000](https://helsinkisolutionoffice.atlassian.net/browse/UHF-0000)

## What was done
Migrations were created using migration API to move old content from xml feeds to the new Drupal website.

* This thing was fixed

## How to install

* Make sure your instance is up and running on correct branch.
  * `git pull origin UHF-0000_insert_correct_branch`
  * `make fresh`
* Run `make drush-cr`

The foloowing drush commands will be available and has to be run in this order:
```

drush migrate:import newsitem_term
drush migrate:import newsitem_image_fi
drush migrate:import newsitem_media_fi
drush migrate:import newsitem_paragraph_fi
drush migrate:import newsitem_node_fi
drush migrate:import newsitem_paragraph_en
drush migrate:import newsitem_node_en
drush migrate:import newsitem_paragraph_sv
drush migrate:import newsitem_node_sv
```

## How to test

* [ ] Check that this feature works
* [ ] Check that code follows our standards

## Designers review

* [ ] This PR does not need designers review
* [ ] This PR has been visually reviewed by a designer (Name of the designer)

## Other PRs

* Link to other PR

### CODE REVIEW

**Review by tapsal — CHANGES_REQUESTED** [review]

File: conf/cmi/core.extension.yml [hunk 1]
```diff
@@ -90,6 +91,8 @@ module:
   metatag_open_graph: 0
   metatag_twitter_cards: 0
   migrate: 0
+  migrate_plus: 0
+  migrate_tools: 0
```
[the remark this answers is not in the document]
Comment (TudorCCalin) [reply conf/cmi/core.extension.yml:96]: Agree, this is to be used from drush.

File: public/modules/custom/helfi_migration/config/install/migrate_plus.migration.newsitem_image_fi.yml [hunk 2]
```diff
@@ -0,0 +1,77 @@
+# Migration configuration for newsitem content.
```
[the remark this answers is not in the document]
Comment (TudorCCalin) [reply public/modules/custom/helfi_migration/config/install/migrate_plus.migration.newsitem_image_fi.yml]: done! moved them to migrations

File: public/modules/custom/helfi_migration/src/Plugin/migrate/source/NewsitemNode.php [hunk 3]
```diff
@@ -0,0 +1,64 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\helfi_migration\Plugin\migrate\source;
+
+use Drupal\migrate\Plugin\MigrationInterface;
+use Drupal\migrate_plus\Plugin\migrate\source\Url;
+use Drupal\migrate\Row;
```
[the remark this answers is not in the document]
Comment (TudorCCalin) [reply public/modules/custom/helfi_migration/src/Plugin/migrate/source/NewsitemNode.php]: done

File: public/modules/custom/helfi_migration/config/install/migrate_plus.migration_group.newsitem.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
+# A "migration group" is - surprise! - a group of migrations. It is used to
```
[the remark this answers is not in the document]
Comment (TudorCCalin) [reply public/modules/custom/helfi_migration/config/install/migrate_plus.migration_group.newsitem.yml]: some comments were removed

File: public/modules/custom/helfi_migration/src/Plugin/migrate/source/NewsitemNode.php [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    while (TRUE) {
+      $pageUrl = $url . "?pageSize=100&page=" . $page;
+      $response = $client->request($method, $pageUrl, $options);
+      $code = $response->getStatusCode();
+      if ($code == 200) {
+        $xmlString = $response->getBody()->getContents();
+        $xmlObject = new \SimpleXMLElement($xmlString);
+        $contentIds = $xmlObject->xpath('/atom:feed/atom:entry/atom:id');
+        foreach ($contentIds as $contentId) {
+          $urlList[] = self::ATOM_NEWS_SOURCE_URL . ((string)$contentId);
```
[the remark this answers is not in the document]
Comment (TudorCCalin) [reply public/modules/custom/helfi_migration/src/Plugin/migrate/source/NewsitemNode.php]: fixed

File: public/modules/custom/helfi_migration/src/Plugin/migrate/source/NewsitemNode.php [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        }
+      }
+      if (count($contentIds) < 100) {
```
[the remark this answers is not in the document]
Comment (TudorCCalin) [reply public/modules/custom/helfi_migration/src/Plugin/migrate/source/NewsitemNode.php]: created a PAGE_SIZE constant

File: public/modules/custom/helfi_migration/src/Plugin/migrate/source/NewsitemNode.php [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+  private function populateUrlList(array &$urlList, string $url) {
+    $method = 'GET';
+    $options = [];
+
+    $client = \Drupal::httpClient();
+    $page = 1;
+    while (TRUE) {
+      $pageUrl = $url . "?pageSize=100&page=" . $page;
+      $response = $client->request($method, $pageUrl, $options);
```
[the remark this answers is not in the document]
Comment (TudorCCalin) [reply public/modules/custom/helfi_migration/src/Plugin/migrate/source/NewsitemNode.php]: done

**Review by tapsal — APPROVED** [review]

**Pull request merged**